src/sql/azdata.proposed.d.ts

```diff
@@ -3918,6 +3918,31 @@ declare module 'azdata' {
 	 * Namespace for interacting with query editor
 	*/
 	export namespace queryeditor {
+		export type QueryEvent =
+			| 'queryStart'
+			| 'queryStop'
+			| 'executionPlan';
+
+		export interface QueryEventListener {
+			onQueryEvent(type: QueryEvent, document: queryeditor.QueryDocument, args: any);
+		}
+
+		// new extensibility interfaces
+		export interface QueryDocument {
+			providerId: string;
+
+			uri: string;
```

Comment:
fileUri?

Reply:
it looks like VS Code is using just `uri` in their `TextDocument` so I think that makes more sense for us as well in this case.